lib/dotcom/alerts/endpoint_stops.ex

```diff
@@ -0,0 +1,155 @@
+defmodule Dotcom.Alerts.EndpointStops do
+  @moduledoc """
+  Module for determining endpoint-stop ranges affected by an alert
+  """
+
+  alias Alerts.Alert
+  alias Dotcom.Alerts.EndpointStops.Behaviour
+
+  @route_patterns_repo Application.compile_env!(:dotcom, :repo_modules)[:route_patterns]
+  @routes_repo Application.compile_env!(:dotcom, :repo_modules)[:routes]
+  @stops_repo Application.compile_env!(:dotcom, :repo_modules)[:stops]
+
+  @behaviour Behaviour
+
+  @impl Behaviour
+  def endpoint_stops(alerts, route_ids) do
+    alerts
+    |> Enum.map(&endpoint_stops_for_alert(&1, route_ids))
+    |> Enum.reject(&Kernel.is_nil/1)
+  end
+
+  defp endpoint_stops_for_alert(alert, route_ids) do
+    affected_stop_ids = alert |> Alert.get_entity(:stop)
+    affected_direction_id = alert |> affected_direction_id()
+
+    direction_names =
+      route_ids
+      |> as_actual_route_ids()
```

Comment:
Technically `Routes.Repo.green_line()` returns a thing you can get direction_names from, FYI

Reply:
Huh - interesting. [That does clean this up somewhat](https://github.com/mbta/dotcom/pull/2558/commits/e685444bc5cb3114ba8449ed30b1e623d729e4cf)